exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DefaultSqlHandler.java

```diff
@@ -237,12 +237,8 @@ protected DrillRel convertToRawDrel(final RelNode relNode) throws SqlUnsupported
       // HEP for rules, which are failed at the LOGICAL_PLANNING stage for Volcano planner
       final RelNode setOpTransposeNode = transform(PlannerType.HEP, PlannerPhase.PRE_LOGICAL_PLANNING, relNode);
 
-      // HEP Join Push Transitive Predicates
-      final RelNode transitiveClosureNode =
-          transform(PlannerType.HEP, PlannerPhase.TRANSITIVE_CLOSURE, setOpTransposeNode);
-
-      // HEP Directory pruning .
-      final RelNode pruned = transform(PlannerType.HEP_BOTTOM_UP, PlannerPhase.DIRECTORY_PRUNING, transitiveClosureNode);
+      // HEP Directory pruning.
+      final RelNode pruned = transform(PlannerType.HEP_BOTTOM_UP, PlannerPhase.DIRECTORY_PRUNING, setOpTransposeNode);
```

Comment:
I recall that back when I had reviewed the original transitive closure PR, one of the things I had asked was why not do the TC before the DIRECTORY_PRUNING phase such that new predicates that are created as a result of TC would be used for pruning.   So, now it seems that doing the TC too early is causing a problem, is that right ?   Does that mean that going forward we won't support the scenario whether an newly inferred directory predicate (e.g dir0 = 2015) would be used for directory pruning  ?

Reply:
You have asked it for `SetOpTranspose` rules, but it is topical for TC too.

Actually directory pruning is performed by two rules: `DirPruneScanFilterOnProjectRule` and `DirPruneScanFilterOnScanRule`.
They matches for EnumerableTableScan and performed in `DIRECTORY_PRUNING` stage and matches for DrillScanRel and performed in `PARTITION_PRUNING` stage. 

In current implementation the TC is used before `PARTITION_PRUNING` stage, so directory pruning performed correctly. 
I have added test case to verify it.

Reply:
@vdiravka thx for the explanation and for adding the test for TC done before partition pruning.